### PR TITLE
automatically publish to dockerhub on circleci

### PR DESCRIPTION
**Pull request by argvk**
pushes to the docker hub repo https://cloud.docker.com/u/contentful/repository/docker/contentful/coredns-nodecache with
* `:latest` tag when merged to master
* `:<branchname>` when pushed

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -7,14 +13,35 @@ jobs:
...
-      docker_layer_caching: true
+    executor: docker-publisher
     steps:
       - checkout
-      - run: make build 
+      - run: make build
+      - run:
+          name: publish docker image with branch
+          command: |
+            docker tag contentful-labs/coredns-nodecache:latest $IMAGE_NAME:$CIRCLE_BRANCH
```
Comment (yannh) [inline .circleci/config.yml]: we only want to tag latest if we are in the master branch?

File: .circleci/config.yml [hunk 2]
```diff
@@ -7,14 +13,35 @@ jobs:
...
+            echo "$docker_password" | docker login -u "$docker_login" --password-stdin
+            docker push $IMAGE_NAME:$CIRCLE_BRANCH
+  publish-latest:
+    executor: docker-publisher
+    steps:
+      - run:
+          name: publish docker image with latest tag 
+          command: |
+            docker tag contentful-labs/coredns-nodecache:latest $IMAGE_NAME:latest
```
Comment (yannh) [inline .circleci/config.yml]: can we push both master & the actual tag? master won't always be repulled

Comment (yannh) [reply .circleci/config.yml]: mhhh actually it will have a sha anyway so that should be fine...

File: .circleci/config.yml (hunk 1 above)
Comment (argvk) [reply .circleci/config.yml]: yes that is how this works, see https://github.com/contentful-labs/coredns-nodecache/pull/11/files#diff-1d37e48f9ceff6d8030570cd36286a61R46-R47

**Pull request merged**